### PR TITLE
This commit completes Task 6 - GET /api/articles/:article_id/comments

### PR DESCRIPTION
**Pull request by yaiqbal**

### CODE REVIEW

File: controllers/news.controller.js [hunk 1]
```diff
@@ -1,36 +1,56 @@
...
-      res.status(200).send({ articles });
+  selectArticles().then((articles) => {
+    res.status(200).send({ articles });
+  });
+};
+
+exports.getCommentsById = async (req, res, next) => {
+  const { article_id } = req.params;
+
+  checkArticleExists(article_id)
```
Comment (nc-poppy-m) [inline controllers/news.controller.js:43]: Lovely function! However, article_id and author are foreign keys, so PSQL will throw an error (that you can handle with a [PSQL error handler](https://notes.northcoders.com/courses/js-back-end/error-handling#psql-thrown-errors)) if you try and insert a non-existent or invalid value.

This means you don't need to check separately (comment these checks out and run the code to see what I mean - you have one PSQL error code covered in your error handler)

File: __tests__/index.test.js [hunk 2]
```diff
@@ -106,4 +106,48 @@ describe('/api/articles', () => {
...
+  });
+  test.only('GET:200 Responds with an empty array of comments for an article_id which exists in the article table but has no comments in the comments table', () => {
+    return request(app)
+      .get('/api/articles/2/comments')
+      .expect(200)
+      .then((response) => {
+        expect(response.body.comments.length).toBe(0)
+      });
+  });
+  test.only('GET:404 sends an appropriate status and error message when given a valid but non-existent article id', () => {
```
Comment (nc-poppy-m) [inline __tests__/index.test.js:137]: Lovely differentiation for these tests between 404 and 400

**Review by nc-poppy-m — COMMENTED** [review]
Great work! See my comments about tests and error handling - you don't need to resubmit the PR for this, but I would suggest implementing the changes to the test. You can also refactor to rely on PSQL errors as this would showcase your understanding of them, although this isn't strictly necessary

File: __tests__/index.test.js [hunk 3]
```diff
@@ -106,4 +106,48 @@ describe('/api/articles', () => {
         });
       });
   });
+})
+
+describe('/api/articles/:article_id/comments', () => {
+  test.only('GET:200 Responds with an array of comments for an article_id which exists in the article table', () => {
```
[the remark this answers is not in the document]
Comment (yaiqbal) [reply __tests__/index.test.js:112]: .only left in error,  removed them. This change will be part of the commit for Task 7.

**Pull request merged**